scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/constraints/activities/ActivityExpression.java

```diff
@@ -614,4 +616,76 @@ public String prettyPrint(final String prefix) {
   @Override
   public void extractResources(final Set<String> names) { }
 
+  //true if second is a subset or equal to first
+  public static boolean subsetOrEqual(SerializedValue first, SerializedValue second){
+    final var visitor = new SerializedValue.Visitor<Boolean>(){
+      @Override
+      public Boolean onNull() {
+        return true;
+      }
+
+      @Override
+      public Boolean onNumeric(final BigDecimal value) {
+        final var firstAsMap = first.asNumeric();
+        if(firstAsMap.isEmpty()){
+          return false;
+        }
+        return firstAsMap.get().equals(value);
+      }
+
+      @Override
+      public Boolean onBoolean(final boolean value) {
+        final var firstAsBoolean = first.asBoolean();
+        if(firstAsBoolean.isEmpty()){
+          return false;
+        }
+        return firstAsBoolean.get().equals(value);
+      }
+
+      @Override
+      public Boolean onString(final String value) {
+        final var firstAsString = first.asString();
+        if(firstAsString.isEmpty()){
+          return false;
+        }
+        return firstAsString.get().equals(value);
```

Comment:
I think we do. Why not ?

Reply:
You do bring up an interesting point - which is that there is a notion of "match" that may be more general than "is syntactically equal to". That may be an avenue worth exploring

Reply:
> especially since activity parameters can make their way directly into spacecraft commands via command expansion, we should err on the side of strictness rather than permissiveness with respect to user input

That's a good point. I'm alright with us erring on the side of strictness as long as we have it documented. So in this case, if we're going with "String comparisons are case-sensitive", that should be in some user-accessible documentation related to this feature.